sale_coupon_most_expensive/models/sale_order.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import models
+
+
+class SaleOrder(models.Model):
+    """Extend to implement most_expensive_product disc option."""
+
+    _inherit = "sale.order"
+
+    def _get_most_expensive_line(self):
+        # Mimic _get_cheapest_line method.
+        lines = self.order_line.filtered(
+            lambda r: not r.is_reward_line and r.price_reduce > 0
```

Comment:
Please use float_compare on price_reduce

Reply:
is it really necessary?

This method mimics:

```
    def _get_cheapest_line(self):
        # Unit prices tax included
        return min(self.order_line.filtered(lambda x: not x.is_reward_line and x.price_reduce > 0), key=lambda x: x['price_reduce'])
```

Here `float_compare` is not used either. Also if it would be used, then you need to specify precision. And what precision would be used here then?

Reply:
@rousseldenis Fixed! Can you update your review please?